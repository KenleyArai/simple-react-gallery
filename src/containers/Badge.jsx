import React, { Component } from "react";
import PropTypes from "prop-types";

import BadgeStyle from "styled/BadgeStyle";

export default class Badge extends Component {
  static propTypes = {
    type: PropTypes.string,
    url: PropTypes.string
  };

  constructor(props) {
    super(props);

    this.state = { type: undefined, url: undefined };
  }

  componentWillMount() {
    this.setState({ type: this.props.type, url: this.props.url });
  }

  render() {
    var { type, url } = { ...this.props };
    switch (type) {
      case "facebook":
        return <BadgeStyle facebook />;
      case "linkedin":
        return <BadgeStyle linkedin />;
      default:
        return <BadgeStyle blank />;
    }
  }
}
