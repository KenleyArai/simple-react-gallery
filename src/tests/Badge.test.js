import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
import "jest-styled-components";

configure({ adapter: new Adapter() });

import Badge from "containers/Badge";

describe("Badge", () => {
  it("should render blank badge with no props", () => {
    const wrapper = shallow(<Badge />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should render linkedin badge with linkedin props", () => {
    const wrapper = shallow(<Badge linkedin />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should render facebook badge with facebook props", () => {
    const wrapper = shallow(<Badge facebook />);
    expect(wrapper).toMatchSnapshot();
  });
});
