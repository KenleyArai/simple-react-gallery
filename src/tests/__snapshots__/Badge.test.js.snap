// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should render blank badge with no props 1`] = `
<styled.div
  blank={true}
/>
`;

exports[`Badge should render facebook badge with facebook props 1`] = `
<styled.div
  blank={true}
/>
`;

exports[`Badge should render linkedin badge with linkedin props 1`] = `
<styled.div
  blank={true}
/>
`;
